kgforge/core/commons/sparql_query_rewriter.py

```diff
@@ -0,0 +1,173 @@
+import re
+from typing import Any, Dict, List, Match, Optional, Tuple, Union, Type
+
+from kgforge.core.commons.context import Context
+from kgforge.core.commons.exceptions import QueryingError
+
+
+# FIXME: need to find a comprehensive way (different than list) to get all SPARQL reserved clauses
+SPARQL_CLAUSES = [
+    "where",
+    "filter",
+    "select",
+    "union",
+    "limit",
+    "construct",
+    "optional",
+    "bind",
+    "values",
+    "offset",
+    "order by",
+    "prefix",
+    "graph",
+    "distinct",
+    "in",
+    "as",
+    "base",
+    "prefix",
+    "reduced",
+    "describe",
+    "ask",
+    "named",
+    "asc",
+    "desc",
+    "from",
+    "optional",
+    "graph",
+    "regex",
+    "union",
+    "str",
+    "lang",
+    "langmatches",
+    "datatype",
+    "bound",
+    "sameTerm",
+    "isIRI",
+    "isURI",
+    "isBLANK",
+    "isLITERAL",
+    "group",
+    "by",
+    "order",
+    "minus",
+    "not",
+    "exists"
+]
+
+
+def rewrite_sparql(query: str, context: Context, metadata_context: Context) -> str:
```

Comment:
This can go to the SPARQLQueryBuilder.

Reply:
Just pushed some changes to put them all in the builder